automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/LeonardoTestUtils.scala

```diff
@@ -291,4 +291,40 @@ trait LeonardoTestUtils extends WebBrowserSpec with Matchers with Eventually wit
     // Return the test result, or throw error
     testResult.get
   }
+
+  def verifyHailImport(notebookPage: NotebookPage, vcfPath: GcsPath, clusterName: ClusterName): Unit = {
+    val welcomeToHail =
+      """Welcome to
+        |     __  __     <>__
+        |    / /_/ /__  __/ /
+        |   / __  / _ `/ / /
+        |  /_/ /_/\_,_/_/_/""".stripMargin
+
+    val vcfSummary =
+      """Samples: 1092
+        |        Variants: 855026
+        |       Call Rate: 0.983877
+        |         Contigs: ['20']
+        |   Multiallelics: 0
+        |            SNPs: 824953
+        |            MNPs: 0
+        |      Insertions: 12227
+        |       Deletions: 17798
+        | Complex Alleles: 48
+        |    Star Alleles: 0
+        |     Max Alleles: 2""".stripMargin
+
+    notebookPage.executeCell("from hail import *") shouldBe None
+    notebookPage.executeCell("hc = HailContext(sc)").get should include(welcomeToHail)
+
+    notebookPage.executeCell(s"chr20vcf = '${vcfPath.toUri}'") shouldBe None
+    notebookPage.executeCell("imported = hc.import_vcf(chr20vcf)").get should include("Hail: INFO: Coerced almost-sorted dataset")
+
+    notebookPage.executeCell("imported.summarize().report()").get should include(vcfSummary)
+
+    // show that the Hail log contains jobs that were run on preemptible nodes
+
+    val preemptibleNodePrefix = clusterName.string + "-sw"
+    notebookPage.executeCell(s"! grep Finished ~/hail.log | grep $preemptibleNodePrefix").get should include(preemptibleNodePrefix)
```

Comment:
:-D

I continue to be amazed that we can write a test that executes a grep command in a notebook cell to check a log for a condition.